### PR TITLE
Feature: AWS Cognito + Azure AD

### PR DESCRIPTION
**Pull request by yolanv**
### Thank you for your contribution! Please go through this small checklist to prepare your pull request to be reviewed as soon as possible :)
* [X] All authors were added to `_data/authors.yml` with a picture and an up-to-date bio
* [X] The blog post image is of ratio 500x293
  * Feel free to leave a comment if you don't know how to resize your image and someone will gladly help you out
* [X] The [**one sentence per line** method](https://raw.githubusercontent.com/brandon-rhodes/blog/master/texts/brandon/2012/one-sentence-per-line.rst) has been adopted
* [X] A spell check has been done
* [X] All links open in a new tab
  * Example: `[Amazing tech blog](https://ordina-jworks.github.io){:target="_blank" rel="noopener noreferrer"}`
* [X] The post's content properly resizes according to the window's size
  * Tip: Just shorten your window's horizontal length to easily test this
  * To have images properly resize use `class="image fit"`
  * To have YouTube videos properly resize, wrap the iframe into a `<div class="responsive-video">`
* [X] If a table of content is: check whether all links are working correctly

### Any specific remarks for the reviewers or anything you need help with?
\<enter text here>

### Checklist for the reviewers before merging
* [ ] Publishing date of the post has been correctly set
  * This can be done by renaming the filename or by setting the date in the header section of the post

### CODE REVIEW

**Comment by yolanv** [comment]
Ready for review :)
@BikersDelight @pietervincken @DenMette

File: _posts/2023-09-18-aws-cognito-azure-ad.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+comments: true
+---
+# Table of contents
+{:.no_toc}
+- TOC
+{:toc}
+# Introduction
+In today's interconnected digital landscape, enterprises are increasingly relying on a mix of cloud solutions to meet their diverse needs.
+One common challenge is efficiently managing user identities and authentication across these platforms.
+But what happens when you're already invested in Azure Active Directory as your primary identity provider?
```
Comment (pietervincken) [inline _posts/2023-09-18-aws-cognito-azure-ad.md:17]: Add that many companies already rely on office 365 and therefor have Azure AD setup (or have legacy on premise)

File: _posts/2023-09-18-aws-cognito-azure-ad.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+While many companies utilize Azure AD, they may not fully adopt the entire Azure ecosystem.
```
Comment (pietervincken) [inline _posts/2023-09-18-aws-cognito-azure-ad.md:19]: Maybe make the O365 link here.

File: _posts/2023-09-18-aws-cognito-azure-ad.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+However, based on our experience, integrating Cognito with AWS services offers a cleaner solution, especially if your application operates on AWS.
```
Comment (pietervincken) [inline _posts/2023-09-18-aws-cognito-azure-ad.md]: Rephrase to flow better. 

Implementing Single Sign-On is a security best-practice, highlight why (easy on-off boarding, central management, easy policy enforcement, ...). 

SSO in many organisations is done through AD (Azure or on-premise) due to O365. 

Implementing SSO in other places makes a ton of sence.

File: _posts/2023-09-18-aws-cognito-azure-ad.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+Although an AWS Lambda function could authenticate with Azure AD, AWS Cognito provides a convenient approach to incorporate authentication within AWS services like API Gateway and CloudFront.
+
+In this article, we'll explore how to efficiently integrate users from your Azure AD environment into AWS Cognito.
+Can these two seemingly distinct systems work together to provide a unified authentication experience?
```
Comment (pietervincken) [inline _posts/2023-09-18-aws-cognito-azure-ad.md:28]: Nice teaser 😄

File: _posts/2023-09-18-aws-cognito-azure-ad.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+  function_name    = "lambda-add-user-to-groups"
+  handler          = "index.handler"
+  role             = aws_iam_role.user_to_group_lambda_iam_role.arn
+  
+  source_code_hash = data.archive_file.user_to_group_lambda_file.output_base64sha256
+  
+  runtime          = "nodejs18.x"
+}
+
+resource "" "user_to_group_lambda_permission" {
```
Comment (pietervincken) [inline _posts/2023-09-18-aws-cognito-azure-ad.md]: Typo? Empty resource type here.

File: _posts/2023-09-18-aws-cognito-azure-ad.md [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+This is why we are storing the ID in the Parameter Store, enabling us to retrieve it whenever necessary.
+```terraform
+resource "aws_ssm_parameter" "ssm_parameter" {
+  name  = "example-application-for-aws-user-pool-id"
+  type  = "String"
+  value = aws_cognito_user_pool.cognito_user_pool.id
+}
+```
+# Conclusion
+By following this approach, we can seamlessly authenticate our enterprise users across AWS services.
```
Comment (pietervincken) [inline _posts/2023-09-18-aws-cognito-azure-ad.md]: In conclusion explain what this has solved compared to alternatives. Integration with other AWS services is a key factor, explain this in more detail. 

Explain what can be improved on the example to make it production ready, if anything. 

Maybe as a final parting thought (optional) add that this is only for allowing applications to authenticate against cognito, it's not to setup SSO for AWS itself.

**Review by pietervincken — CHANGES_REQUESTED** [review]

**Comment by pietervincken** [comment]
Interesting post and super relevant for me 😄  Got a question about this today in my project. So I will share the post as soon as it's live!

**Review by pietervincken — APPROVED** [review]

**Comment by yolanv** [comment]
Wilt er iemand nog een review doen of is het oké zo? @DenMette  @BikersDelight

**Comment by denmette** [comment]
> Wilt er iemand nog een review doen of is het oké zo? @DenMette @BikersDelight

Voor mij is het goed, dan lanceren we hem binnen 5 dagen.

**Review by denmette — APPROVED** [review]

**Pull request merged**